src/coreComponents/constitutive/fluid/BlackOilFluid.hpp

```diff
@@ -19,57 +19,1281 @@
+                      arraySlice2d< real64, multifluid::USD_PHASE_DC - 2 > const & dPhaseVisc_dGlobalCompFrac ) const;
+
+  GEOSX_HOST_DEVICE
+  void computeRs( real64 Pb,
+                  real64 & Rs,
+                  real64 & dRs_dP ) const;
+
+  GEOSX_HOST_DEVICE
+  void computeSaturatedBoVisc( real64 Rs,
+                               real64 dRs_dP,
+                               real64 & Bo,
+                               real64 & dBo_dP,
+                               real64 & visc,
+                               real64 & dVisc_dP )  const;
+
+  GEOSX_HOST_DEVICE
+  void computeUndersaturatedBoVisc( bool needDerivs,
+                                    localIndex numHydrocarbonComp,
+                                    real64 P,
+                                    real64 Rs,
+                                    real64 const dRs_dC[],
+                                    real64 & Bo,
+                                    real64 & dBo_dP,
+                                    real64 dBo_dC[],
+                                    real64 & visc,
+                                    real64 & dVisc_dP,
+                                    real64 dvisc_dC[] ) const;
+
+  GEOSX_HOST_DEVICE
+  void computeUndersaturatedBoVisc( real64 Rs,
+                                    real64 P,
+                                    real64 & Bo,
+                                    real64 & visc ) const;
+
+  GEOSX_HOST_DEVICE
+  void computeMassMoleDensity( bool needDerivs,
+                               bool useMass,
+                               localIndex numHydrocarbonComp,
+                               real64 Rs,
+                               real64 dRs_dP,
+                               real64 const dRs_dC[],
+                               real64 Bo,
+                               real64 dBo_dP,
+                               real64 const dBo_dC[],
+                               real64 & dens,
+                               real64 & dDens_dP,
+                               arraySlice1d< real64, multifluid::USD_PHASE_DC - 3 > const & dDens_dC ) const;
+
+  /// Phase ordering info
+  arrayView1d< integer const > m_phaseTypes;
+  arrayView1d< integer const > m_phaseOrder;
+  arrayView1d< integer const > m_hydrocarbonPhaseOrder;
+
+  /// Surface mass density for each phase
+  arrayView1d< real64 const > m_surfacePhaseMassDensity;
+
+  /// Table kernel wrappers to interpolate in the oil and gas (B vs p) tables
+  arrayView1d< TableFunction::KernelWrapper const > m_formationVolFactorTables;
+
+  /// Table kernel wrappers to interpolate in the oil and gas (\mu vs p) tables
+  arrayView1d< TableFunction::KernelWrapper const > m_viscosityTables;
+
+  /// Water reference pressure
+  real64 const m_waterRefPressure;
+
+  /// Water formation volume factor
+  real64 const m_waterFormationVolFactor;
+
+  /// Water compressibility
+  real64 const m_waterCompressibility;
+
+  /// Water viscosity
+  real64 const m_waterViscosity;
```

Comment:
There's no need to make member variables `const`. All the compute/update functions are already `const` so there's no danger of accidentally modifying a member.